app/routes/datasets/versions.py

```diff
@@ -232,13 +231,18 @@ async def append_to_version(
     # For the background task, we only need the new source uri from the request
     input_data = {"creation_options": deepcopy(default_asset.creation_options)}
     input_data["creation_options"]["source_uri"] = request.source_uri
+    # check if layers are provided and append to creation options
+    if input_data["creation_options"].get("layers"):
+        input_data["creation_options"]["layers"] = request.layers
```

Comment:
Shouldn't the `if` be testing the value of request.layers (to be non-None), rather than the value of layers that came from the default asset?

With the current code, if the existing default asset has some layers, and request.layers is None (no layers specified), then you will set input_data["creation_options"]["layers"] to None, but I think you want to keep it at the current layers, right?  (Hopefully I'm understanding the semantics of pydantic fields correctly.)